packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -607,6 +631,20 @@ export async function downloadSnapshot(
 		});
 	}
 
+	if (loadingGroupIds !== undefined) {
+		// If we need to fetch snapshot using group Ids but no ids are specified, then fetch
+		// full snapshot using "default" groupId.
+		if (fetchFullSnapshot) {
```

Comment:
What does fetchFullSnapshot == true && loadingGroupIds == [] mean?
Or worse, fetchFullSnapshot == true && loadingGroupIds == ["some value"]?

Passing data as two semi-related fields is dangerous as it creates these states that could happen, and thus needs to manage. It's like using 2 booleans to pass 3-state - it's much better (if possible) to use enum with 3 values.

Reply:
Resolving as per offline discussion.